src/main/java/seedu/address/logic/commands/schedulercommands/EditEventCommand.java

```diff
@@ -1,21 +1,150 @@
 package seedu.address.logic.commands.schedulercommands;
 
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_EVENTS;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
 import seedu.address.logic.commands.Command;
 import seedu.address.logic.commands.CommandResult;
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
+import seedu.address.model.event.Event;
+import seedu.address.model.event.EventName;
+import seedu.address.model.event.EventTime;
+import seedu.address.model.tag.Tag;
 
 public class EditEventCommand extends Command {
 
     public static final String COMMAND_WORD = "editevent";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the Event identified "
+            + "by the index number used in the displayed Event list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_NAME + "NAME] "
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_NAME + "CS2103T homework"
+            + PREFIX_DATE + "1-2-2020 1259";
+    public static final String MESSAGE_EDIT_EVENT_SUCCESS = "Edited Event: %1$s";
+    public static final String MESSAGE_DUPLICATE_EVENT = "This event already exists in the Event list.";
+
+    private Index index;
+    private EditEventDescriptor descriptor;
+
+    /**
+     * Creates an EditEventCommand.
+     *
+     * @param index index of the event.
+     * @param descriptor the container that hold the changes.
+     */
+    public EditEventCommand(Index index, EditEventDescriptor descriptor) {
+        requireNonNull(index);
+        requireNonNull(descriptor);
+        this.index = index;
+        this.descriptor = descriptor;
+    }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
-        return null;
+        requireNonNull(model);
+        List<Event> lastShownList = model.getFilteredEventList();
+        //TODO: need to catch index out of bound error somewhere
```

Comment:
If im not wrong, index out of bounds error message is stored under commons messages, can refer to the other features to see how it has been checked.

Reply:
ok thanks!